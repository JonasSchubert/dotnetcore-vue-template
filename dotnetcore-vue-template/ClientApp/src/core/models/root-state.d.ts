import { ExamplesState } from '@/examples/models';
import { BaseState } from "@/libs/core/models";
import { I18nState } from "@/libs/i18n/models";
import { LocalStorageState } from "@/libs/local-storage/models";

export interface RootState extends BaseState {
  examples?: ExamplesState;
  i18n?: I18nState;
  localStorage?: LocalStorageState;
}
