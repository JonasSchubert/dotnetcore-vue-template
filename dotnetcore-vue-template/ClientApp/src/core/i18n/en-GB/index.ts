/* eslint-disable quote-props */
export default {
  'message': {
    'add-failure': 'Failed to add {0}',
    'add-success': 'Successfully added {0}',
    'app-name': '.NetCore Vue Template',
    'cancel': 'Cancel',
    'dark-theme': 'Dark theme',
    'date-time-added': 'Added on',
    'date-time-updated': 'Updated on',
    'de-DE': 'German',
    'de-DE-native': 'Deutsch',
    'delete': 'Delete',
    'delete-failure': 'Failed to delete {0}',
    'delete-message': 'Do you really want to delete {0}?',
    'delete-success': 'Successfully deleted {0}',
    'delete-title': 'Delete?',
    'en-GB': 'English',
    'en-GB-native': 'English',
    'es-ES': 'Spanish',
    'es-ES-native': 'Español',
    'example': 'Example | Examples',
    'home': 'Home | Homes',
    'id': 'Id | Ids',
    'last-used': 'Last used',
    'light-theme': 'Light theme',
    'search': 'Search',
    'ta-LK': 'Tamil',
    'ta-LK-native': 'தமிழ்',
    'update-failure': 'Failed to update {0}',
    'update-success': 'Successfully updated {0}',
    'value': 'Value | Values'
  }
};
