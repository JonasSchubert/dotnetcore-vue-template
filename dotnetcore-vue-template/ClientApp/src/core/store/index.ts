import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { createModule as createExamplesModule } from '@/examples/store';
import { ModuleType as ExamplesModule } from '@/examples/store/types';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      [ExamplesModule]: createExamplesModule(),
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('dotnetcore-vue-template', vuetify)
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
