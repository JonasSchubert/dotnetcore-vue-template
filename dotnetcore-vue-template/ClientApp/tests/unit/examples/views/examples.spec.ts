import { shallowMount } from '@vue/test-utils';
import i18n from '@/core/i18n';
import vuetify from '@/core/plugins/vuetify.plugin';
import { createStore } from '@/core/store';
import { Example } from '@/examples/models';
import Examples from '@/examples/views/Examples.vue';

const store = createStore(vuetify);

// Mount the component
const wrapper = shallowMount(Examples, {
  i18n,
  store,
  vuetify
});

describe('Examples', () => {
  describe('computed', () => {
    describe('headers', () => {
      test('should not change unexpected', () => {
        // Arrange && Act
        const { headers } = wrapper.vm as any;

        // Assert
        expect(headers).toMatchSnapshot();
      });
    });
  });

  describe('methods', () => {
    describe('onDeleteConfirmed', () => {
      test('should call $store.dispatch', () => {
        // Arrange
        const item: Example = {
          id: 2,
          value: 42
        } as unknown as Example;
        store.dispatch = jest.fn();

        // Act
        (wrapper.vm as any).onDeleteConfirmed(item);

        // Assert
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith('examples/delete', { item });
      });
    });

    describe('requestDelete', () => {
      test('should call $refs.deleteConfirmDialog.open', () => {
        // Arrange
        const item: Example = {
          id: 2,
          value: 42
        } as unknown as Example;
        (wrapper.vm as any).$refs = {
          deleteConfirmDialog: { open: jest.fn() }
        };

        // Act
        (wrapper.vm as any).requestDelete(item);

        // Assert
        expect((wrapper.vm as any).$refs.deleteConfirmDialog.open).toHaveBeenCalledTimes(1);
        expect((wrapper.vm as any).$refs.deleteConfirmDialog.open).toHaveBeenCalledWith({
          buttons: {
            cancel: {
              text: "message.cancel"
            },
            confirm: {
              text: "message.delete",
              textColor: "error"
            }
          },
          message: "message.delete-message",
          messageParams: ["42"],
          meta: item,
          title: "message.delete-title"
        });
      });
    });
  });
});
