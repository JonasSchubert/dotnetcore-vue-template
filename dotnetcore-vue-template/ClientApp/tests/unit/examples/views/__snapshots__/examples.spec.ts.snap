// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples computed headers should not change unexpected 1`] = `
Array [
  Object {
    "sortable": false,
    "text": "Id",
    "value": "id",
  },
  Object {
    "sortable": true,
    "text": "Value",
    "value": "value",
  },
  Object {
    "sortable": true,
    "text": "Added on",
    "value": "dateTimeAdded",
  },
  Object {
    "sortable": true,
    "text": "Updated on",
    "value": "dateTimeUpdated",
  },
  Object {
    "sortable": false,
    "text": "",
    "value": "actions",
  },
]
`;
